server/core/store.go

```diff
@@ -640,7 +663,7 @@ func (s *StoresInfo) SetRegionSize(storeID uint64, regionSize int64) {
 // UpdateStoreStatus updates the information of the store.
 func (s *StoresInfo) UpdateStoreStatus(storeID uint64, leaderCount int, regionCount int, pendingPeerCount int, leaderSize int64, regionSize int64) {
 	if store, ok := s.stores[storeID]; ok {
-		newStore := store.Clone(SetLeaderCount(leaderCount),
+		newStore := store.ShallowClone(SetLeaderCount(leaderCount),
```

Comment:
Can you briefly list some optimization points made in this PR and their benefits? like optimization of precheck can improve 50%, optimization of store clone can improve xx%

Reply:
OK,I wil give the list tomorrow.

Reply:
Because the log and storage will affect the test results, during the comparison process, the log and storage were commented out by me. The test conditions are: 1 million region 20store 0.1 region-update-ratio, 10heartbeat-rounds. The average heartbeat is the average of the last 9 times.

the improve percent like this:
no.|improve content |related code |average heartbeat/s | improve percent
---|---|---|---|---
1|   master branch |     |   133051 | 0%    
2|  not remove regin in the tree when old region is in the tree  |  region.go 633 : oldRegionItem.region = region   |  138364  |     4%
3|  not remove regin in the subtree when old region's peer is not change  is in the subtree  |  region.go 635-647 : ```if remove {r.removeRegionFromSubTree(origin)}```   |  175809  |     27%
4|PreCheckPutRegion add the 'LOOSE' check.|  cluster.go 447: origin, _, _, err := c.core.PreCheckPutRegion(region, core.LOOSE)   | 182601   |     3.86%
5|use ShallowClone when UpdateStoreStatus.|   store.go 666: newStore := store.ShallowClone(SetLeaderCount(leaderCount),  | 240027   |     31%
6| reduce the number of updates when update related stores.   |  cluster.go 575:		var storeMap map[uint64]uint64  |  244630  |     1.86%

Reply:
thanks! 👍

Reply:
Would you like split the PR to multiple PR, It's easy to review, reduce bugs, and even speed up your PR to merge. and also we can more easy to trace why do the change.

Reply:
OK.PTAL.The first one : #2389